File: final_export_to_downloads.py
from PIL import Image
import os
import sys

# 本物のソース画像パスを指定 (カズ様の完璧な素材 + AIのフィーチャー画像)
image_paths = [
    r"c:\antigravity\AppStore_PERFECT\1_Home_Perfect.jpg",
    r"c:\antigravity\AppStore_PERFECT\2_Player_Perfect.jpg",
    r"c:\antigravity\AppStore_PERFECT\3_Generate_Perfect.jpg",
    r"c:\antigravity\AppStore_PERFECT\4_Marketing_Subconscious_Perfect.jpg",
    r"c:\antigravity\AppStore_PERFECT\5_Marketing_Philosophy_Perfect.jpg",
    r"C:\Users\kamek\.gemini\antigravity\brain\9d8d6805-eb6d-40d0-884a-5216fe515f20\app_feature_graphic_1774950217732.png"
]

# リサイズ設定
target_width = 1284
target_height = 2778
feature_width = 1024
feature_height = 500

# 出力先を C:\PlayStore_Android に固定
output_dir = r"C:\PlayStore_Android"

if not os.path.exists(output_dir):
    os.makedirs(output_dir)

file_names = [
    "1_Home_fixed.png",
    "2_Player_fixed.png",
    "3_Generate_fixed.png",
    "4_Marketing_Subconscious_fixed.png",
    "5_Marketing_Philosophy_fixed.png",
    "Feature_Graphic_AI.png"
]

print(f"--- Pythonによる画像リサイズ開始 ---")
print(f"出力先: {output_dir}")

for i, path in enumerate(image_paths):
    if not os.path.exists(path):
        print(f" [!] ファイルが見つかりません: {path}")
        continue
        
    try:
        print(f" [{i+1}/6] 処理中: {os.path.basename(path)}")
        img = Image.open(path)
        
        # フィーチャーグラフィックだけサイズが異なる
        if i == 5:
            w, h = feature_width, feature_height
        else:
            w, h = target_width, target_height
            
        # リサイズ（LANCZOSは高品質）
        img_resized = img.resize((w, h), Image.Resampling.LANCZOS)
        
        output_path = os.path.join(output_dir, file_names[i])
        # PNGとして保存
        img_resized.save(output_path, "PNG")
        print(f"  -> 保存完了: {output_path}")
    except Exception as e:
        print(f"  [!] エラー発生: {str(e)}")

print("--- すべての画像生成が完了しました！ ---")

File: final_export_to_downloads.js
const { Jimp } = require('jimp');
const path = require('path');
const fs = require('fs');

/**
 * Jimp v1.6.x 完全対応版
 * 1. resizeの引数を数値で確実に渡す
 * 2. writeがPromiseを返すのを正しくawaitする
 */

const image_paths = [
    "C:\\antigravity\\AppStore_PERFECT\\1_Home_Perfect.jpg",
    "C:\\antigravity\\AppStore_PERFECT\\2_Player_Perfect.jpg",
    "C:\\antigravity\\AppStore_PERFECT\\3_Generate_Perfect.jpg",
    "C:\\antigravity\\AppStore_PERFECT\\4_Marketing_Subconscious_Perfect.jpg",
    "C:\\antigravity\\AppStore_PERFECT\\5_Marketing_Philosophy_Perfect.jpg",
    "C:\\Users\\kamek\\.gemini\\antigravity\\brain\\9d8d6805-eb6d-40d0-884a-5216fe515f20\\app_feature_graphic_1774950217732.png"
];

const file_names = [
    "1_Home_fixed.png",
    "2_Player_fixed.png",
    "3_Generate_fixed.png",
    "4_Marketing_Subconscious_fixed.png",
    "5_Marketing_Philosophy_fixed.png",
    "Feature_Graphic_AI.png"
];

const output_dir = "C:\\PlayStore_Android";

async function run() {
    console.log(`--- アセット生成開始 (Jimp v1.6対応版) ---`);
    
    if (!fs.existsSync(output_dir)) {
        fs.mkdirSync(output_dir, { recursive: true });
    }

    for (let i = 0; i < image_paths.length; i++) {
        const src = image_paths[i];
        const dest = path.join(output_dir, file_names[i]);
        
        try {
            console.log(`[${i+1}/6] 処理中: ${path.basename(src)}`);
            const image = await Jimp.read(src);
            
            // 数値として確実に渡す
            const w = i === 5 ? 1024 : 1284;
            const h = i === 5 ? 500 : 2778;

            console.log(`    リサイズ実行: ${w}x${h}`);
            // Jimp v1.6 の修正: オブジェクトで指定
            image.resize({ width: w, height: h });
            
            console.log(`    ファイルを書き出し中: ${dest}`);
            // Jimp v1.6 の write は Promise を返す
            await image.write(dest);
            console.log(`    完了!`);
        } catch (e) {
            console.error(`    [!] エラー: ${src}`);
            console.error(e);
        }
    }
    
    console.log("--- 全画像生成完了！ ---");
}

run();
